### PR TITLE
gh-266 ChainedIterable made public

### LINKED ISSUE #266: ChainedIterable Duplicate Class
linked to : [ChainedIterable Duplicated Classes. #2369](https://github.com/gchq/Gaffer/issues/2369)

Make ChainedIterable public

### PR DESCRIPTION
**Pull request by GCHQDev404**
# Related Issue

- Resolve #266

### CODE REVIEW

File: core/src/main/java/uk/gov/gchq/koryphe/iterable/ChainedIterator.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.Iterator;
+
+/**
+ * @param <T> the type of items in the iterator
+ */
+public class ChainedIterator<T> implements Closeable, Iterator<T> {
+    private final Iterator<? extends Iterable<? extends T>> iterablesIterator;
+    private Iterator<? extends T> currentIterator = Collections.emptyIterator();
+
+    public ChainedIterator(final Iterator<? extends Iterable<? extends T>> iterablesIterator) {
```
Comment (r32575) [inline core/src/main/java/uk/gov/gchq/koryphe/iterable/ChainedIterator.java:31]: The Gaffer version of `ChainedIterable` has another constructor

File: core/src/main/java/uk/gov/gchq/koryphe/iterable/ChainedIterable.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import java.io.Closeable;
+import java.util.Iterator;
+
+/**
+ * @param <T> the type of items in the iterator
+ */
+public class ChainedIterable<T> implements Closeable, Iterable<T> {
+    private final Iterable<? extends Iterable<? extends T>> iterables;
+
+    public ChainedIterable(final Iterable<? extends Iterable<? extends T>> iterables) {
```
Comment (r32575) [inline core/src/main/java/uk/gov/gchq/koryphe/iterable/ChainedIterable.java:30]: The Gaffer version of `ChainedIterable` has another constructor. If you are going to get rid of the Gaffer equivalent and point at the Koryphe version then you may want to add it

**Comment by r32575** [comment]
There are tests for `ChainedIterable` in Gaffer you may want to copy across. I don't know about the other iterables...

File: core/src/main/java/uk/gov/gchq/koryphe/iterable/ChainedIterable.java (hunk 2 above)
Comment (t92549) [reply core/src/main/java/uk/gov/gchq/koryphe/iterable/ChainedIterable.java:30]: Good shout, I made them public but forgot to look at Gaffer and move across any constructors or tests

**Review by r32575 — APPROVED** [review]

**Pull request merged**